explorer_frontend/tests/tutorials/tutorials.test.mts

```diff
@@ -0,0 +1,117 @@
+
+import type { CheckProps } from "../../src/features/tutorial-check/CheckProps";
+import runTutorialCheckOne from "../../src/features/tutorial-check/checks/tutorialOneCheck";
+import runTutorialCheckΤwo from "../../src/features/tutorial-check/checks/tutorialTwoCheck";
+import runTutorialCheckThree from "../../src/features/tutorial-check/checks/tutorialThreeCheck";
+import runTutorialCheckFour from "../../src/features/tutorial-check/checks/tutorialFourCheck";
+import { deploymentEffect, RPC_ENDPOINT, setCompletedTutorialEvent, setTutorialChecksEvent, tutorialContractStepFailed, tutorialContractStepPassed } from "./globals";
+import { expect, describe, test } from "vitest";
+const solc = require("solc");
+import path from "node:path";
+import { createCompileInput } from "../../src/features/shared/utils/solidityCompiler/helper.ts";
+import type { App } from "../../src/features/code/types.ts";
+import runTutorialCheckFive from "../../src/features/tutorial-check/checks/tutorialFiveCheck.ts";
+
+const TEST_PROPS: CheckProps = {
+  rpcUrl: RPC_ENDPOINT,
+  deploymentEffect: deploymentEffect,
+  setTutorialChecksEvent: setTutorialChecksEvent,
+  tutorialContractStepFailed: tutorialContractStepFailed,
+  tutorialContractStepPassed: tutorialContractStepPassed,
+  contracts: [],
+  setCompletedTutorialEvent: setCompletedTutorialEvent
+};
+
+const createContracts = async (code: string) => {
+  const input = createCompileInput(code);
+  const res = JSON.parse(solc.compile(JSON.stringify(input)));
+
+  const contracts: App[] = [];
+  if ("contracts" in res && res.contracts !== undefined && "Compiled_Contracts" in res.contracts) {
+    for (const name in res.contracts?.Compiled_Contracts) {
+      const contract = res.contracts.Compiled_Contracts[name];
+
+      contracts.push({
+        name: name,
+        bytecode: `0x${contract.evm.bytecode.object}`,
+        sourcecode: code,
+        abi: contract.abi,
+      });
+    }
+  }
+
+  return contracts;
+}
+
+
+describe("Tutorial One tests", async () => {
+  test("Tutorial One passes with the given solution", async () => {
+    const code = await import(path.resolve(__dirname, "../../src/features/tutorial/assets/tutorialOne/tutorialOneSolution.sol?raw"));
+    const codeRes = code.default;
+
+    const contracts = await createContracts(codeRes);
+    TEST_PROPS.contracts = contracts;
```

Comment:
if you want test propos. create new object with extension it would be more safe way. `const testProps = {...testPropos, contracts };`